src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/integration/LegacyResettlementAssessmentIntegrationTest.kt

```diff
@@ -0,0 +1,319 @@
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.Pathway
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.ResettlementAssessmentRequest
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.ResettlementAssessmentStatus
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.Status
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.resettlementassessment.ResettlementAssessmentRequestQuestionAndAnswer
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.resettlementassessment.StringAnswer
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.jpa.entity.PrisonerEntity
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.jpa.entity.ResettlementAssessmentEntity
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.jpa.entity.ResettlementAssessmentQuestionAndAnswerList
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.jpa.entity.ResettlementAssessmentSimpleQuestionAndAnswer
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.jpa.entity.ResettlementAssessmentType
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.jpa.repository.PathwayStatusRepository
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.jpa.repository.ResettlementAssessmentRepository
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+class LegacyResettlementAssessmentIntegrationTest : IntegrationTestBase() {
+
+  @Autowired
+  private lateinit var resettlementAssessmentRepository: ResettlementAssessmentRepository
+
+  @Autowired
+  private lateinit var pathwayStatusRepository: PathwayStatusRepository
+
+  // Tests using ATTITUDES_THINKING_AND_BEHAVIOUR Pathway - i.e. using old AbstractResettlementAssessmentStrategy
+  @Test
+  @Sql("classpath:testdata/sql/seed-pathway-statuses-2.sql")
+  fun `Post get next assessment page happy path`() {
+    val nomsId = "G1458GV"
+    val pathway = "ATTITUDES_THINKING_AND_BEHAVIOUR"
+    val assessmentType = "BCST2"
+    val currentPage = "HELP_TO_MANAGE_ANGER"
+    val questions = mutableListOf<ResettlementAssessmentRequestQuestionAndAnswer<*>>(
+      ResettlementAssessmentRequestQuestionAndAnswer("HELP_TO_MANAGE_ANGER", answer = StringAnswer("NO")),
+    )
+    val body = ResettlementAssessmentRequest(
+      questionsAndAnswers = questions,
+    )
+
+    webTestClient.post()
+      .uri("/resettlement-passport/prisoner/$nomsId/resettlement-assessment/$pathway/next-page?assessmentType=$assessmentType&currentPage=$currentPage")
+      .bodyValue(body)
+      .headers(setAuthorisation(roles = listOf("ROLE_RESETTLEMENT_PASSPORT_EDIT")))
+      .exchange()
+      .expectStatus().isOk
+      .expectHeader().contentType("application/json")
+      .expectBody().json(
+        """
+        {
+          "nextPageId": "ISSUES_WITH_GAMBLING"
+        }
+        """.trimIndent(),
+      )
+  }
+
+  @Test
```

Comment:
I understand most of these tests are integration tests, quite difficult to understand (for someone like me) although the functionality implemented here could be unit tested quite easily, I don't know if it is an advantage or disadvantage to have this imbalance of integration tests vs unit going forward...just a consideration